### PR TITLE
Adding some network and security enhancements for vpc environment

### PR DESCRIPTION
**Pull request by oleksiiRybak**
The goal of this PR is to put certain networking policies in order and separate the back-end part (DB services) from the publicly accessible front-end part (business logic services) so that they can be managed more separately with different security policies.

### CODE REVIEW

File: deploy/terraform/variables.tf [hunk 1]
```diff
@@ -500,6 +500,12 @@ variable "memory" {
   default = 8192
 }
 
+variable "tcp_network_protocol" {
```
Comment (adrianclay) [inline deploy/terraform/variables.tf:503]: Does this make sense to be a variable?  I can't see this wanting to change 🤔.

File: deploy/terraform/main.tf [hunk 2]
```diff
@@ -180,8 +180,8 @@ resource "aws_ecs_service" "gp2gp_ps" {
 
   network_configuration {
     subnets = [for subnet in aws_subnet.nia_gp2gp_public_subnet : subnet.id]
-    assign_public_ip = "true"
-    security_groups = [aws_security_group.nia_gp2gp_dmz.id, aws_security_group.ps_db_migration.id]
+    assign_public_ip = true
```
Comment (adrianclay) [inline deploy/terraform/main.tf:183]: Does assigning a public IP still make sense if we have the load balancers now?

File: deploy/terraform/network.tf [hunk 3]
```diff
@@ -178,17 +193,17 @@ resource "aws_security_group" "nia_gp2gp_mq" {
...
 
   ingress {
     description     = "Allow PS DB traffic"
-    from_port       = "5432"
+    from_port       = var.PS_DB_PORT
     protocol        = "tcp"
-    to_port         = "5432"
-    security_groups = [aws_security_group.nia_gp2gp_dmz.id]
+    to_port         = var.PS_DB_PORT
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (adrianclay) [inline deploy/terraform/network.tf:206]: Is allowing connectivity from anywhere advisable?

File: deploy/terraform/variables.tf (hunk 1 above)
Comment (oleksiiRybak) [reply deploy/terraform/variables.tf:503]: yes, I think it is still makes sense as it looks more concise. Also, in some places you can spot some upper case strings in some other ones lower cases hence having it as a variable will be more concise.

File: deploy/terraform/network.tf (hunk 3 above)
Comment (oleksiiRybak) [reply deploy/terraform/network.tf:206]: yes, but this doesn't work (the connection from aws_security_group.nia_gp2gp_dmz.id) with this configuration hence I had to change it all though it is not open to external connections, I tested it. For DB we are using private subnet so it doesn't mean that it is open to all.

Comment (oleksiiRybak) [reply deploy/terraform/network.tf:206]: this is a place for further deeper analysis but at the moment it seems to be working OK now, the DB is not accessible for external access.

Comment (oleksiiRybak) [reply deploy/terraform/network.tf:206]: I can add an additional ticket to see how this cidr "0.0.0.0/0" block can be more limited to connections from nia_gp2gp_dmz security group only.

**Review by adrianclay — APPROVED** [review]

**Pull request merged**